jax/experimental/export/serialization.py

```diff
@@ -386,17 +386,24 @@ def _deserialize_sharding(s: ser_flatbuf.Sharding) -> export.Sharding:
 
 
 def _serialize_effect(builder: flatbuffers.Builder, eff: core.Effect) -> int:
-  # TODO(necula): for now serialize just the name of the class
   try:
-    _ = eff.__class__()
+    eff_replica = eff.__class__()
   except:
     raise NotImplementedError(
-        f"serializing effect {eff} that does not have a nullary class"
-        " constructor"
+        f"Effect {eff} must have a nullary constructor to be serializable"
+    )
+  try:
+    hash_eff = hash(eff)
+    hash_eff_replica = hash(eff_replica)
+  except:
```

Comment:
Consider doing `except Exception`. Bare `except` also handles exceptions like `KeyboardInterrupt` and `GeneratorExit`, which are not meant to be handled (in most contexts).

Reply:
Done